doc/source/reference/c-api/iterator.rst

```diff
@@ -203,6 +203,67 @@ is used to control the memory layout of the allocated result, typically
     }
 
 
+Multi Index Tracking Example
+----------------------------
+
+This example shows you how to work with the :c:data:`NPY_ITER_MULTI_INDEX` flag. For simplicity, we assume the argument is a two-dimensional array.
+
+.. code-block:: c
+
+   int PrintMultiIndex(PyArrayObject *arr) {
+       NpyIter *iter;
+       NpyIter_IterNextFunc *iternext;
+       npy_intp *multi_index;
```

Comment:
```suggestion
       npy_intp multi_index[2];
```
You know that the index is 2-D, so this is fine.  Otherwise, a typical pattern would be using `NPY_MAXDIMS`.